### PR TITLE
replace nxs with h5py in SANSUtility

### LINKED ISSUE #19796: Remove nexus python dependency
[SANSUtility.py](https://github.com/mantidproject/mantid/blob/master/scripts/SANS/SANSUtility.py) uses both `h5py` and the nexus python api (`nxs`). Change the calls to the latter to use the former (i.e. everything should use `h5py`). The bits that need to be redone are to remove lines 30-36, then redo the functions [`can_load_as_event_workspace`](https://github.com/mantidproject/mantid/blob/master/scripts/SANS/SANSUtility.py#L1523) and [`get_measurement_time`](https://github.com/mantidproject/mantid/blob/master/scripts/SANS/SANSUtility.py#L1768).

**Note:** Make sure to branch off of `release-v3.10` so it can be merged into the release.

### PR DESCRIPTION
**Pull request by yxqd**
Fixes #19796 

Refs #19799 

**To test:**

Make sure following tests pass
* systemtest SANSFileChecking
* unittest SANSUtility

No release notes required.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/SANSUtility.py [hunk 1]
```diff
@@ -1746,46 +1738,41 @@ class MeasurementTimeFromNexusFileExtractor(object):
...
-    def _check_if_processed_nexus_file(self, nxs_file):
-        nxs_file.opendata('definition')
-        mantid_definition = nxs_file.getdata()
-        nxs_file.closedata()
-
+    def _get_measurement_time_processed_file(self, h5entry):
+        logs = h5entry['logs']
+        end_time = logs['end_time']
+        value = end_time['value']
+        print("value=", value)
```
Comment (AntonPiccardoSelg) [inline scripts/SANS/SANSUtility.py]: Why do you print here?

File: scripts/SANS/SANSUtility.py [hunk 2]
```diff
@@ -1746,46 +1738,41 @@ class MeasurementTimeFromNexusFileExtractor(object):
...
+        return value[0]
+
+    def _get_measurement_time_for_non_processed_file(self, h5entry):
+        end_time = h5entry['end_time']
+        print("end_time=", end_time)
```
Comment (AntonPiccardoSelg) [inline scripts/SANS/SANSUtility.py]: Same as above

File: scripts/SANS/SANSUtility.py [hunk 3]
```diff
@@ -1746,46 +1738,41 @@ class MeasurementTimeFromNexusFileExtractor(object):
...
+        return end_time[0]
+
+    def _check_if_processed_nexus_file(self, h5entry):
+        definition = h5entry['definition']
+        mantid_definition = definition[0]
+        print("mantid_definition=", mantid_definition)
```
Comment (AntonPiccardoSelg) [inline scripts/SANS/SANSUtility.py]: Same as above

**Review by AntonPiccardoSelg — CHANGES_REQUESTED** [review]

File: scripts/SANS/SANSUtility.py [hunk 4]
```diff
@@ -1535,23 +1527,23 @@ def can_load_as_event_workspace(filename):
...
-                rootKeys =  list(nxs_file.getentries().keys())
-                nxs_file.opengroup(rootKeys[0])
-                nxs_file.opengroup('event_workspace')
-                is_event_workspace = True
-            except:
-                pass
-            finally:
-                nxs_file.close()
-
+    # Original code test it with nexus python. why?
```
Comment (AntonPiccardoSelg) [inline scripts/SANS/SANSUtility.py]: What is the question?

**Comment by AntonPiccardoSelg** [comment]
@yxqd Is this PR essential for M3.10?

**Comment by AntonPiccardoSelg** [comment]
I ran `SANSFileChecking` on a Python3 build and it seems to fail for me:
```
Traceback (most recent call last):
File "/home/anton/sources/Mantid3/mantid/Testing/SystemTests/tests/analysis/SANSFileChecking.py", li
ne 180, in test_that_reducer_for_LARMOR_switches_to_correct_IDF_when_outdated_V2
self.assertNotEqual(os.path.normpath(idf_workspace), os.path.normpath(idf_reducer_before))
AssertionError: '/home/anton/sources/Mantid3/mantid/instrument/LARMOR_Definition.xml' == '/home/anton/
sources/Mantid3/mantid/instrument/LARMOR_Definition.xml'
```

I think the might be an issue with `utf-8` encoding.

**Comment by yxqd** [comment]
Thanks @AntonPiccardoSelg. The bugs are fixed. Please check

**Review by AntonPiccardoSelg — APPROVED** [review]

**Comment by AntonPiccardoSelg** [comment]
* Code changes look fine
* Tests pass on Python3
* I ran reductions with SANS2D and LARMOR with event-mode data and multi-period event-mode data. This touches parts of the code changes and it seemed fine.
* I ran reductions with LOQ and can confirm that switching of the IDF works fine (this is anyway thoroughly covered by the `SANSFileChecking` tests)

The change is still risky, but at this point there is not more we can do. Hence, please :shipit:

**Pull request merged**